pkg/merger/merger_test.go

```diff
@@ -408,6 +408,71 @@ func Test_MergeAndUpdate(t *testing.T) {
 		})
 	}
 }
+func Test_RecordLastModified(t *testing.T) {
+	cases := []struct {
+		name   string
+		attrs  *storage.ReaderObjectAttrs
+		mets   *Metrics
+		source string
+	}{
+		{
+			name:   "nil attrs; succeeds",
+			attrs:  nil,
+			mets:   &Metrics{},
+			source: "fake-source",
+		},
+		{
+			name:   "nil mets; succeeds",
+			attrs:  &storage.ReaderObjectAttrs{},
+			mets:   nil,
+			source: "fake-source",
+		},
+		{
+			name:   "nil; succeeds",
+			attrs:  nil,
+			mets:   nil,
+			source: "",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			recordLastModified(tc.attrs, tc.mets, tc.source)
+		})
+	}
+}
+
+func Test_RecordLastModified_RecordMetrics(t *testing.T) {
+	var fakeMetric FakeInt64
+
+	attrs := &storage.ReaderObjectAttrs{
+		LastModified: time.Now(),
+	}
+	mets := &Metrics{
+		LastModified: &fakeMetric,
+	}
+	source := ""
+
+	recordLastModified(attrs, mets, source)
+	if fakeMetric.last != 0 {
+		t.Errorf("recordLastModified() fail; got %d, want %d", fakeMetric.last, 0)
+	}
+}
+
+type FakeInt64 struct {
```

Comment:
What other packages are you exporting `FakeInt64` to?

If no one, then please do not export it: `fakeInt64`